### ISSUE TITLE
usage question:

### ISSUE THREAD

**Issue by johngrabner** [bojone/accum_optimizer_for_keras#2]
Hi Jianlin Su,

Thank you for this accumulator. I suspect I don't understand something.
I'm expecting similar results between these 2 experiments:

**1) with your accumulator (I divide batch size by 2, accumulate across 2 batch)**
```
img_gen = generator(minibatch_size = minibatch_size/2) 
optimizer = keras.optimizers.SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True, clipnorm=5)
opt = accum_optimizer.AccumOptimizer(optimizer, 2)
model.compile(loss=keras.losses.binary_crossentropy, optimizer=opt, metrics=["accuracy"])
```

**Run 1:**
---------
```
Epoch 1/100 loss: 0.6930 
Epoch 2/100 loss: 0.6918 
Epoch 3/100 loss: 0.6897 
Epoch 4/100 loss: 0.6901 
Epoch 5/100 loss: 0.6841 
```

**Run 2:**
---------
```
Epoch 1/100 loss: 0.6911 - acc: 0.5279 - val_loss: 0.6785 - val_acc: 0.5632
Epoch 2/100 loss: 0.6898 - acc: 0.5485 - val_loss: 0.6689 - val_acc: 0.5704
Epoch 3/100 loss: 0.6876 - acc: 0.5471 - val_loss: 0.6712 - val_acc: 0.6172
Epoch 4/100 loss: 0.6900 - acc: 0.5418 - val_loss: 0.6936 - val_acc: 0.4876
Epoch 5/100 loss: 0.6883 - acc: 0.5476 - val_loss: 0.6661 - val_acc: 0.6184
```

**2) without your accumulator (nominal batch size, no accumulation)**
```
img_gen = generator(minibatch_size = minibatch_size)
optimizer = keras.optimizers.SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True, clipnorm=5)
model.compile(loss=keras.losses.binary_crossentropy, optimizer=optimizer, metrics=["accuracy"])
```

**Run 1:**
---------
```
Epoch 1/100 loss: 0.6039 
Epoch 2/100 loss: 0.4983 
Epoch 3/100 loss: 0.4373 
Epoch 4/100 loss: 0.4184 
Epoch 5/100 loss: 0.3928 
```

**Run 2:**
---------

```
Epoch 1/100 loss: 0.6640 
Epoch 2/100 loss: 0.5702 
Epoch 3/100 loss: 0.4759 
Epoch 4/100 loss: 0.4389 
Epoch 5/100 loss: 0.4094 
```

**Question:**

If I divide the batch by half and accumulate over 2, should I not see similar results?

**Comment by johngrabner** [comment]
I tried with mse

mse with accum and batch / 2

Epoch 1/100 loss: 0.2486 
Epoch 2/100 loss: 0.2482 
Epoch 3/100 loss: 0.2479 
Epoch 4/100 loss: 0.2422 
Epoch 5/100 loss: 0.2393

mse without accum, full batch

Epoch 1/100 loss: 0.2448 
Epoch 2/100 loss: 0.2194 
Epoch 3/100 loss: 0.1886 
Epoch 4/100 loss: 0.1697 
Epoch 5/100 loss: 0.1524 

So very different even with mse.

**Comment by johngrabner** [comment]
**mse, adam with accum, half batch**

Epoch 1/100 loss: 0.3008
Epoch 2/100 loss: 0.2760
Epoch 3/100 loss: 0.2782
Epoch 4/100 loss: 0.2899
Epoch 5/100 loss: 0.2758



**mse, adam without accum, full batch**

Epoch 1/100 loss: 0.2490 
Epoch 2/100 loss: 0.1858 
Epoch 3/100 loss: 0.1504 
Epoch 4/100 loss: 0.1319 
Epoch 5/100 loss: 0.1239

same as before, the loss does not improve as much when using accumulation.

**Comment by johngrabner** [comment]
I tried other experiments and looks like accumulation works.
